contracts/XVSVault/XVSVault.sol

```diff
@@ -348,10 +343,20 @@ contract XVSVault is XVSVaultStorage, ECDSA {
         require(user.amount >= user.pendingWithdrawals.add(_amount), "requested amount is invalid");
 
         PoolInfo storage pool = poolInfos[_rewardToken][_pid];
+
+        _updatePool(_rewardToken, _pid);
```

Comment:
Nice. So if I understand correctly this was moved to follow the checks-effects-interaction pattern?

Reply:
We moved it because now we are going to pay rewards in `requestWithdrawal` so for that `_updatePool` needs to be called before calculating rewards.